app_dart/lib/src/request_handlers/vacuum_github_commits.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:gcloud/db.dart';
+import 'package:github/github.dart';
+import 'package:meta/meta.dart';
+import 'package:truncate/truncate.dart';
+
+import '../datastore/config.dart';
+import '../model/appengine/commit.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../service/datastore.dart';
+import '../service/github_service.dart';
+import '../service/scheduler.dart';
+
+/// Query GitHub for commits from the past day and ensure they exist in datastore.
+@immutable
+class VacuumGithubCommits extends ApiRequestHandler<Body> {
+  const VacuumGithubCommits(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @required this.scheduler,
+    @visibleForTesting this.datastoreProvider = DatastoreService.defaultProvider,
+  })  : assert(datastoreProvider != null),
+        super(config: config, authenticationProvider: authenticationProvider);
+
+  final DatastoreServiceProvider datastoreProvider;
+
+  final Scheduler scheduler;
+
+  @override
+  Future<Body> get() async {
+    final DatastoreService datastore = datastoreProvider(config.db);
+
+    for (String repository in Config.schedulerSupportedRepos) {
+      final String owner = repository.split('/').first;
```

Comment:
Should we start saving the owner and repo separately?

Reply:
I like the idea of these being stored as a `RepositorySlug` as then it's simple to pull owner, repo, and fullName. I'll follow up with this refactor in a separate PR - https://github.com/flutter/flutter/issues/80531